### PR TITLE
ui: Reset state for workflow item history when changing tabs

### LINKED ISSUE #317: ui: Workflowitem history crashes
Steps to reproduce: 
- Go to a subproject page
- Click on the button to show workflow item details
- Click on "History" 
- Click on another tab, e.g. "Documents" 
- Click on "History" again
- You are logged out due to a bad request

Possible issue: When changing tabs, the state is not cleared and therefore the offset and limit are not correctly calculated.

### PR DESCRIPTION
**Pull request by mathiashoeld**
# Changes
When a user views the workflow item details and clicks on history and
then back to another tab, the history state is now reset, so the
calculation of offset and limit are correct.

Add e2e test that makes sure this does not happen again.

Refactor e2e tests, so that 'data-test' is used instead of 'id'.

# Related issues
Closes #317

# How to test
When viewing a workflow item's history, there should be no issues.

### CODE REVIEW

File: frontend/src/pages/Workflows/WorkflowDetails.js [hunk 1]
```diff
@@ -144,6 +145,12 @@ function WorkflowDetails({
     },
     [showWorkflowDetails]
   );
+  useEffect(
```
Comment (gonzochic) [inline frontend/src/pages/Workflows/WorkflowDetails.js]: The effect is called every time, even if it is not necessary. I believe this hook has to move into the component itself.

File: frontend/src/pages/WorkflowitemDetails/reducer.js [hunk 2]
```diff
@@ -35,7 +35,8 @@ export default function reducer(state = initialState, action) {
         currentHistoryPage: action.currentHistoryPage,
         isHistoryLoading: false
       });
-
+    case CLOSE_WORKFLOWITEM_HISTORY:
+      return initialState;
     case CLOSE_WORKFLOWITEM_DETAILS:
```
[the remark this answers is not in the document]
Comment (mathiashoeld) [reply frontend/src/pages/WorkflowitemDetails/reducer.js]: done

File: frontend/src/pages/Workflows/WorkflowDetails.js (hunk 1 above)
Comment (mathiashoeld) [reply frontend/src/pages/Workflows/WorkflowDetails.js]: changed implementation of `<WorkflowitemHistoryTab>` to class based and used `componentWillUnmount` to reset state

File: frontend/src/pages/WorkflowitemDetails/reducer.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (mathiashoeld) [reply frontend/src/pages/WorkflowitemDetails/reducer.js]: also added the `LOGOUT` action here

File: frontend/src/pages/Workflows/WorkflowitemHistoryTab/WorkflowHistoryTab.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
+import React from "react";
+import { connect } from "react-redux";
+
+import { toJS } from "../../../helper";
+import ScrollingHistory from "../../Common/History/ScrollingHistory";
+import { fetchNextWorkflowitemHistoryPage, resetWorkflowitemHistory } from "./actions";
+
+class WorkflowitemHistoryTab extends React.Component {
+  constructor(props) {
```
[the remark this answers is not in the document]
Comment (mathiashoeld) [reply frontend/src/pages/Workflows/WorkflowitemHistoryTab/WorkflowHistoryTab.js]: done

**Pull request merged**